### ISSUE TITLE
Cone/Cone and Cylinder/Cylinder collisions are not working when using ODECollisionDetector

### ISSUE THREAD

**Issue by traversaro** [conda-forge/dartsim-feedstock#39]
### Comment:

Currently it is disabled as it fails: https://github.com/conda-forge/dartsim-feedstock/pull/38#issuecomment-1554792029 .

**Comment by traversaro** [comment]
Given what I learned in the past two days (https://github.com/conda-forge/gz-physics-feedstock/pull/24 and https://github.com/conda-forge/dartsim-feedstock/pull/60) I renamed the issue, and it is not only a test that was not passing, but an high level behavior difference that users may notice. My main guess is that the issue is due to some details of the libode build in conda-forge, but this needs to be investigated.

fyi @jslee02 this is problably the bug for which you had to disable `testConeCone` and `testCylinderCylinder` in the pixi CI upstream in dart.
